### PR TITLE
Ngio projection

### PR DESCRIPTION
**Pull request by lorenzocerrone**
* Add ngio to dependecies 
* Refactor projection task to use ngio 

## Checklist before merging
- [x] I added an appropriate entry to `CHANGELOG.md`

### CODE REVIEW

File: fractal_tasks_core/tasks/projection.py [hunk 1]
```diff
@@ -55,122 +65,65 @@ def projection(
             `create_cellvoyager_ome_zarr_init`.
     """
     method = DaskProjectionMethod(init_args.method)
-    logger.info(f"{init_args.origin_url=}")
-    logger.info(f"{zarr_url=}")
-    logger.info(f"{method=}")
+    ngio_logger.info(f"{init_args.origin_url=}")
```
Comment (jluethi) [inline fractal_tasks_core/tasks/projection.py:68]: Is the plan to generally transition to using ngio logger in any task that uses ngio?

File: fractal_tasks_core/tasks/projection.py [hunk 2]
```diff
@@ -189,5 +142,5 @@ def projection(
 
     run_fractal_task(
```
Comment (jluethi) [inline fractal_tasks_core/tasks/projection.py:143]: To discuss with @tcompa: When does `run_fractal_task` use the logger name? Is the ngio_logger also a reasonable choice here?

File: fractal_tasks_core/tasks/projection.py [hunk 3]
```diff
@@ -55,122 +65,65 @@ def projection(
...
+    # Copy over the tables
+    for roi_table in original_ngff_image.tables.list(table_type="roi_table"):
+        table = original_ngff_image.tables.get_table(roi_table)
+        mip_table = new_ngff_image.tables.new(
+            roi_table, table_type="roi_table", overwrite=True
         )
 
+        roi_list = []
+        for roi in table.rois:
+            roi.z_length = roi.z + 1
```
Comment (jluethi) [inline fractal_tasks_core/tasks/projection.py:120]: We need to set roi.z = 0, to avoid getting IndexErrors if roi.z was not 0 before. It should be 0 before in all our cases though

**Comment by jluethi** [comment]
This is great @lorenzocerrone !

A few brief questions: Have you tried running this projection task on a zyx dataset (no channel or time dimension)? Could we easily add a synthetic test case for that using an ngio generated tiny zyx OME-Zarr. That then closes https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/840

Also, would we expect this task to already work with time-data, e.g. an tczyx dataset?

TBD how much we should test in tasks-core

Can we add an option to copy over additional tables (e.g. user can deactivate that, but we copy non-roi tables by default). We'd keep them & ignore edge-cases (like missing label images) for now. TODO Joel: provide example dataset of condition tables for this.

TODO for myself: 

- [ ] Check how we need to provide the metadata update for https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/841. Likely a change to the `image_list_update_dict`
- [ ] Run example 01 with this branch, to check for https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/823

**Comment by jluethi** [comment]
When do we test separately in tasks core: Only when there is task logic specifically to those cases

File: .github/workflows/ci_poetry.yml [hunk 4]
```diff
@@ -15,7 +15,7 @@ jobs:
 
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        python-version: ["3.10", "3.11", "3.12"]
```
[the remark this answers is not in the document]
Comment (jluethi) [reply .github/workflows/ci_poetry.yml:18]: Yes, we're planning to drop support for Python 3.9 in fractal-tasks-core

[the remark this answers is not in the document]
Comment (jluethi) [reply .github/workflows/ci_poetry.yml:18]: Let's also drop it in pyproject.toml

File: fractal_tasks_core/tasks/projection.py [hunk 5]
```diff
@@ -12,28 +12,38 @@
...
-)
-from fractal_tasks_core.tables import write_table
-from fractal_tasks_core.tables.v1 import get_tables_list_v1
+
 from fractal_tasks_core.tasks.io_models import InitArgsMIP
 from fractal_tasks_core.tasks.projection_utils import DaskProjectionMethod
-from fractal_tasks_core.zarr_utils import OverwriteNotAllowedError
 
+if TYPE_CHECKING:
+    from ngio.core import Image
```
[the remark this answers is not in the document]
Comment (jluethi) [reply fractal_tasks_core/tasks/projection.py:29]: TBD: Coming in a separate PR if we want this style across tasks core

**Comment by lorenzocerrone** [comment]
> * [ ]  Optionally update the task logger to improve its format

I think this should be handled in a separate PR

**Comment by lorenzocerrone** [comment]
The new version of the task has been tested with:

- `tczyx`
- `tzyx`
- `czyx`
- `zyx`

File: fractal_tasks_core/tasks/projection.py [hunk 6]
```diff
@@ -55,122 +65,65 @@ def projection(
...
-            new_non_ROI_table,
-            table_attrs=old_non_ROI_table_attrs,
-            overwrite=init_args.overwrite,
+    dest_dask = method.apply(dask_array=source_dask, axis=z_axis_index)
+    dest_dask = da.expand_dims(dest_dask, axis=z_axis_index)
+    new_image.set_array(dest_dask)
+    new_image.consolidate()
+    # Ends
+
+    # Copy over the tables
```
[the remark this answers is not in the document]
Comment (lorenzocerrone) [reply fractal_tasks_core/tasks/projection.py:113]: The responsability of copying the tables is now moved to the `NgffImage.derive_new_image` method.
